### DISCUSSION TITLE
Show how to do more general types of selection (balancing, diverging etc.)

### DISCUSSION THREAD

**Discussion by gregorgorjanc** [gaynorr/AlphaSimR#146] [General]
We usually simulate directional selection.

In some cases, we might want to simulate balancing selection or diverging selection etc. While these types of selection can be done with the `selectInd()` and some scripting, it would be instructive to show how to use `trait = selectionFun(pop) ...` to simulate different types of selection - we just need an appropriate `selectionFun()` ...

**Comment by gaynorr** [comment]
To expand on your comments for the benefit of others, here is an example of how it can currently be done in AlphaSimR with the `abs` function as your selection function. This approach only works for a single trait. For multiple traits, you'd have to write your own selection function. The specifics of how to write such a function haven't really been documented. I'm not sure where best to document this.

As an aside, creating a selection function that returns a single value as a function of multiple traits is a great way to incorporate crop modeling approaches and/or simulate complex epistasis.

```
library(AlphaSimR)

founderPop = runMacs(nInd = 1000, nChr=10, segSites = 1000)

SP = SimParam$
  new(founderPop)$
  addTraitA(nQtlPerChr = 1000)$
  setVarE(h2 = 0.3)

# Perform stabilizing selection, minimize absolute value
pop = newPop(founderPop)
hist( gv(pop) )

gain = numeric(50)
for(gen in 1:50){
  pop = selectCross(pop, nInd = 100, nCrosses = 1000, 
                    trait = abs, selectTop = FALSE)
  
  gain[gen] = mean( abs( gv(pop) ) )
}
hist( gv(pop) )
plot(1:50, gain, type="l", xlab="Generation", ylab="Trait Mean")

# Perform divergent selection, maximize absolute value
# Likely to randomly go to one of the trait extremes
pop = newPop(founderPop)
hist( gv(pop) )

gain = numeric(50)
for(gen in 1:50){
  pop = selectCross(pop, nInd = 100, nCrosses = 1000, 
                    trait = abs)
  
  gain[gen] = mean( abs( gv(pop) ) )
}
hist( gv(pop) )
plot(1:50, gain, type="l", xlab="Generation", ylab="Trait Mean")

```

**Reply by gaynorr** [reply]
My divergent selection example isn't the best, because it will likely just be directional selection in a random direction.

**Comment by gregorgorjanc** [comment]
As to the documenting this behaviour, I think best to add a simple example to `selectInd()` examples section in the docs and/or link to function like `selectionIndex()`. I will open a PR for this.

**Reply by gregorgorjanc** [reply]
PR in https://github.com/gaynorr/AlphaSimR/pull/148
